public/apps/configuration/utils/internal-user-list-utils.tsx

```diff
@@ -0,0 +1,49 @@
+/*
+ *   Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ *   Licensed under the Apache License, Version 2.0 (the "License").
+ *   You may not use this file except in compliance with the License.
+ *   A copy of the License is located at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *   or in the "license" file accompanying this file. This file is distributed
+ *   on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ *   express or implied. See the License for the specific language governing
+ *   permissions and limitations under the License.
+ */
+
+import { map } from 'lodash';
+import { HttpStart } from '../../../../../../src/core/public';
+import { API_ENDPOINT_INTERNALUSERS } from '../constants';
+import { DataObject, InternalUser, ObjectsMessage } from '../types';
+
+export interface InternalUsersListing extends InternalUser {
+  username: string;
+}
+
+export function transformUserData(rawData: DataObject<InternalUser>): InternalUsersListing[] {
+  return map(
+    rawData,
+    (value: InternalUser, key?: string) =>
+      ({
+        username: key,
+        attributes: value.attributes,
+      } as InternalUsersListing)
+  );
+}
+
+export async function requestDeleteUsers(http: HttpStart, users: string[]) {
+  users.forEach(async (r) => {
+    await http.delete(`${API_ENDPOINT_INTERNALUSERS}/${r}`);
+  });
```

Comment:
could we check and ensure this is expected way we want here?

https://codeburst.io/javascript-async-await-with-foreach-b6ba62bbf404

Reply:
Thanks for pointing out. Tried `for...of` and seems works:
~~~javascript
const delay = ms => new Promise(res => setTimeout(res, ms));
for (let num of [1, 2, 3]) {
  delay(1000);
  console.log(num);
}
console.log("Done");

VM499:4 1
VM499:4 2
VM499:4 3
VM499:6 Done
~~~

Will switch to `for...of`

Reply:
Updated.